### DISCUSSION TITLE
Android Status Bar hidden by default on Fullscreen App

### DISCUSSION THREAD

**Discussion by FizixTiramu** [appcelerator/titanium_mobile#14018] [Q&A] [answered]
I feel like I am asking a lot of questions atm. 

This is related to Android fullscreen styling.

I have a full screen app where the window layer fills the entire height and width of the screen and has a background image which is visible below the status and action bars (which are semi-transparent). 

Then I'm using a windowSafeArea layer for screen content which is clamped to the available space below the devices top status bar and above android action bar. This works all correctly.

The only weird thing is that the top status bar on Android with the time and notifications is hidden by default, I have to pull it down from the top of the screen to show it. Then it animates out after a few seconds.

Is there a setting I'm not aware of that is causing this, and can it be controlled? Ideally, what I'd like is for the top status bar to be on screen all of the time, with a completely transparent background.

Window Layer config below:

```
if(Ti.Platform.osname == 'android') {
   windowFlagStr = Ti.UI.Android.FLAG_TRANSLUCENT_NAVIGATION | Ti.UI.Android.FLAG_TRANSLUCENT_STATUS | Ti.UI.Android.SOFT_INPUT_STATE_ALWAYS_VISIBLE;
   windowSoftInputModeStr = Ti.UI.Android.SOFT_INPUT_ADJUST_RESIZE;
}

var windowLayer = Ti.UI.createWindow({
   id: 'windowLayer,
   title: 'windowLayer',
   top: 0,
   left: 0,
   zIndex: 1,
   extendSafeArea: true,
   theme: 'Theme.Titanium.Light.Fullscreen',
   windowFlags: windowFlagStr,
   windowSoftInputMode: windowSoftInputModeStr,
   backgroundImage: '/assets/images/dynamic/appBackgroundImage.png',
});
```

Theme XML Below:
```
<?xml version="1.0" encoding="utf-8"?>
<resources>
	<style name="appCustomTheme" parent="Theme.Titanium.Light.Fullscreen">
		<item name="android:statusBarColor">@android:color/transparent</item>
		<item name="android:windowTranslucentNavigation">false</item>
		<item name="android:navigationBarColor">#ededed</item>
		<item name="android:colorPrimaryDark">#ededed</item>
		<item name="android:windowBackground">#ededed</item>
		<item name="android:colorBackground">#ededed</item>
       </style>
</resources>
```

**Comment by m1ga** [comment]
> is hidden by default, I have to pull it down from the top of the screen to show it

isn't that the point of the fullscreen layout? Maybe the real question is: what kind of layout do you want to have :smile: ?

> I feel like I am asking a lot of questions atm.

don't worry! That's the point of this section to ask and get help.

**Reply by FizixTiramu** [reply]
What I want is for is the background image on the window layer to fill the entire screen and be visible underneath the status bar and nav bar. A fullscreen layout achieves this, so I presume that is what I need?

What I'm trying to achieve is:
1. The window layer to overflow underneath the status and nav bar, so that the background image is visible.
2. The status bar to be transparent and the action bar to be semi-transparent so you can see the background image underneath
3. The status bar to always be visible


See attached screenshot, this is how I want it to appear, however I want the statusbar up top to be visible all of the time on a transparent background.

![Screenshot_20240411-173254](https://github.com/tidev/titanium-sdk/assets/156670741/a781199c-5a60-4199-9afc-9decec720359)

**Reply by m1ga** [reply, marked as answer]
> See attached screenshot, this is how I want it to appear, however I want the statusbar up top to be visible all of the time on a transparent background.

The screenshot show the fullscreen layout, yes. But if you want to have the statusbar be visible it is a different layout. That should be the `no_limits` flag: https://github.com/tidev/titanium-sdk/pull/11911#issuecomment-684111283 and you can use this flag: https://titaniumsdk.com/api/titanium/ui/android.html#flag_layout_no_limits

**Comment by FizixTiramu** [comment]
> > See attached screenshot, this is how I want it to appear, however I want the statusbar up top to be visible all of the time on a transparent background.
> 
> The screenshot show the fullscreen layout, yes. But if you want to have the statusbar be visible it is a different layout. That should be the `no_limits` flag: [#11911 (comment)](https://github.com/tidev/titanium-sdk/pull/11911#issuecomment-684111283) and you can use this flag: https://titaniumsdk.com/api/titanium/ui/android.html#flag_layout_no_limits

I see, I'll try that in a few. Thanks :)

**Comment by FizixTiramu** [comment]
Tested it with theme Theme.Titanium.Light.NoTitleBar and it now appears which is great. However, the bar is semi-transparent not fully transparent. From the links, this is done from a theme but I can't seem to find direction on which one. Any ideas on that?

**Reply by m1ga** [reply]
Sorry, I would need to search for that myself. It's normal native theming stuff so there should be more SO about that I guess.

**Comment by FizixTiramu** [comment]
I've been having a look into it and I'm seeing posts say to set the statusbar colour programmatically, but it seems really inconsistent. So I'm going to leave it for now. Thanks for the help, the important thing is sorted :)
